testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -0,0 +1,440 @@
+					meta.customEvaluation,
+				); err != nil {
+					return err
+				}
+				fmt.Printf("prysm ob: %v/n", beaconPathsAndObjects[path].prysmResps[key])
+				fmt.Printf("lighthouse ob: %v", beaconPathsAndObjects[path].prysmResps[key])
+			case "ssz":
+				sszparams := meta.params("ssz", currentEpoch)
+				if len(sszparams) == 0 {
+					continue
+				}
+				apipath := pathFromParams(path, sszparams)
+				fmt.Printf("ssz api path: %s", apipath)
+				prysmr, lighthouser, err := compareSSZMulticlient(beaconNodeIdx, meta.basepath, apipath)
+				if err != nil {
+					return err
+				}
+				beaconPathsAndObjects[path].prysmResps[key] = prysmr
+				beaconPathsAndObjects[path].lighthouseResps[key] = lighthouser
+			default:
+				return fmt.Errorf("unknown encoding type %s", key)
+			}
+		}
+	}
+	forkPathData := beaconPathsAndObjects["/beacon/states/{param1}/fork"]
+	prysmForkData, ok := forkPathData.prysmResps["json"].(*apimiddleware.StateForkResponseJson)
+	if !ok {
+		return errors.New("failed to cast type")
+	}
+	lighthouseForkData, ok := forkPathData.lighthouseResps["json"].(*apimiddleware.StateForkResponseJson)
+	if !ok {
+		return errors.New("failed to cast type")
+	}
+	if prysmForkData.Data.Epoch != lighthouseForkData.Data.Epoch {
+		return fmt.Errorf("prysm response %v does not match lighthouse response %v",
+			prysmForkData,
+			lighthouseForkData)
+	}
+
+	finalizedEpoch, err := strconv.ParseUint(prysmForkData.Data.Epoch, 10, 64)
+	if err != nil {
+		return err
+	}
+	blockPathData := beaconPathsAndObjects["/beacon/blocks/{param1}"]
+	sszrspL, ok := blockPathData.prysmResps["ssz"].([]byte)
+	if !ok {
+		return errors.New("failed to cast type")
+	}
+	sszrspP, ok := blockPathData.lighthouseResps["ssz"].([]byte)
+	if !ok {
+		return errors.New("failed to cast type")
+	}
+	if finalizedEpoch < 8 {
+		blockP := &ethpb.SignedBeaconBlock{}
+		blockL := &ethpb.SignedBeaconBlock{}
+		if err := blockL.UnmarshalSSZ(sszrspL); err != nil {
+			return errors.Wrap(err, "lighthouse ssz error")
+		}
+		if err := blockP.UnmarshalSSZ(sszrspP); err != nil {
+			return errors.Wrap(err, "prysm ssz error")
+		}
+		if len(blockP.Signature) == 0 || len(blockL.Signature) == 0 || hexutil.Encode(blockP.Signature) != hexutil.Encode(blockL.Signature) {
+			return fmt.Errorf("prysm response %v does not match lighthouse response %v",
+				blockP,
+				blockL)
+		}
+	} else if finalizedEpoch > 8 && finalizedEpoch < 10 {
+		blockP := &ethpb.SignedBeaconBlockAltair{}
+		blockL := &ethpb.SignedBeaconBlockAltair{}
+		if err := blockL.UnmarshalSSZ(sszrspL); err != nil {
+			return errors.Wrap(err, "lighthouse ssz error")
+		}
+		if err := blockP.UnmarshalSSZ(sszrspP); err != nil {
+			return errors.Wrap(err, "prysm ssz error")
+		}
+
+		if len(blockP.Signature) == 0 || len(blockL.Signature) == 0 || hexutil.Encode(blockP.Signature) != hexutil.Encode(blockL.Signature) {
+			return fmt.Errorf("prysm response %v does not match lighthouse response %v",
+				blockP,
+				blockL)
+		}
```

Comment:
This block of code is duplicated here and for Bellatrix fork. I think you should be able to deduplicate it.